neovim: build on aarch64

tkameyama opened the pull request:
Currently,` spack install neovim` is failed to build lua-jit.
neovim installere try to install lua-jit 2.0, but lua-jit is not supported aarch64.

This PR avoid build third-party package on neovim, insted of spack installed packages.

neovim 0.4 need libvterm 0.1 or later.
but current libvterm version is 681 but older then 0.1.
So current version of libvterm is change to 0.0.681.
libvterm is used neovim only.

Commit:
libvterm: renumber version and add 1.0.3
neovim: build on aarrch64

Commit:
Remove unneeded comment.

adamjstewart commented on var/spack/repos/builtin/packages/libvterm/package.py:
```diff
@@ -9,12 +9,23 @@
-    version('681', sha256='abea46d1b0b831dec2af5d582319635cece63d260f8298d9ccce7c1c2e62a6e8')
+    version('0.1.3', preferred=True, sha256='e41724466a4658e0f095e8fc5aeae26026c0726dce98ee71d6920d06f7d78e2b')
+    version('0.0.726', sha256='6344eca01c02e2270348b79e033c1e0957028dbcd76bc784e8106bea9ec3029d')
+    version('0.0.681', sha256='abea46d1b0b831dec2af5d582319635cece63d260f8298d9ccce7c1c2e62a6e8')
```
I wonder if we should even include these bazaar snapshots, I doubt they follow the same versioning scheme.

tkameyama replied:
Changed bazaar snapshot 726 as version 0.0.

adamjstewart replied:
I would just remove that version altogether unless someone needs it.

tkameyama replied:
neovim 0.3.4 or older is needed that verson.

adamjstewart commented on var/spack/repos/builtin/packages/neovim/package.py:
```diff
@@ -26,36 +25,26 @@ class Neovim(CMakePackage):
+    depends_on('msgpack-c', type='link')
+    depends_on('msgpack-c@1.0.0:', type='link', when='@0.4:')
+    depends_on('gperf', type='link')
+    depends_on('libluv@1.30.0:', type='link', when='@0.4:')
```
It looks like many of these dependencies were switched from `@:0.4` to `@0.4:`, was that intentional?

tkameyama replied:
Yes.
On #16410, neovim 0.4 fetch and build dependent packages itself.
So almost package dependency for 0.4 is not needed.

But I build neovim 0.4 on aarch64, neovim fetched package (lua-jit 2.0) dose not build.
So in this PR, the dependent packages is used to spack installed tem.

adamjstewart replied:
Ah, I see. So now the dependency is always required, but some versions have a specific version requirement. That sounds good to me.

Commit:
libvterm:   newer bazaar snapshot version is set to version 0.0.
neovim: change for libvterm version change, and libtermkey version bug is fixed.

adamjstewart commented on var/spack/repos/builtin/packages/libvterm/package.py:
```diff
@@ -9,9 +9,10 @@
+    url      = "http://www.leonerd.org.uk/code/libvterm/libvterm-0.1.3.tar.gz"
 
-    version('681', sha256='abea46d1b0b831dec2af5d582319635cece63d260f8298d9ccce7c1c2e62a6e8')
+    version('0.1.3', preferred=True, sha256='e41724466a4658e0f095e8fc5aeae26026c0726dce98ee71d6920d06f7d78e2b')
```
No need for `preferred` here, it will automatically choose the newest version.

adamjstewart commented on var/spack/repos/builtin/packages/libvterm/package.py:
```diff
@@ -9,9 +9,10 @@
 
-    version('681', sha256='abea46d1b0b831dec2af5d582319635cece63d260f8298d9ccce7c1c2e62a6e8')
+    version('0.1.3', preferred=True, sha256='e41724466a4658e0f095e8fc5aeae26026c0726dce98ee71d6920d06f7d78e2b')
+    version('0.0', sha256='6344eca01c02e2270348b79e033c1e0957028dbcd76bc784e8106bea9ec3029d', url='http://www.leonerd.org.uk/code/libvterm/libvterm-0+bzr728.tar.gz')
```
Let's call it `0.0.0` just to be consistent with the new 3 digit numbering system.

Commit:
update libvterm versions.
